src/lib/shared/form/reducer.ts

```diff
@@ -0,0 +1,103 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { FormActions } from './actions'
+import { Values, FormState } from './types'
+
+const FormReducer = <T extends Values>(state: FormState<T>, action: FormActions<T>): FormState<T> => {
+  switch (action.type) {
+    case 'Form/Submit':
+      return { ...state, isSubmitting: true, error: undefined }
+    case 'Form/SubmitError':
+      return {
+        ...state,
+        isSubmitting: false,
+        error: action.payload,
+      }
+    case 'Form/SubmitSuccess':
```

Comment:
Was this intentionally left empty? After successful submit we set the values again...?

Reply:
fallthrough, same as set values essentially

Reply:
Ok